core/src/main/java/org/apache/accumulo/core/file/rfile/RFile.java

```diff
@@ -426,6 +426,12 @@ public void append(Key key, Value value) throws IOException {
         throw new IllegalArgumentException("Keys appended out-of-order.  New key " + key + ", previous key " + prevKey);
       }
 
+      // 5 bytes of overhead for fields: value, row, colFamily, colQualifier, colVisibility, and 8 for the long timestamp
+      if (((long) key.getSize() + (long) value.getSize() + (5L * 5L + 8L)) >= Integer.MAX_VALUE) {
+        throw new IllegalArgumentException("Key/value pair is too large (" + ((long) key.getSize() + (long) value.getSize())
```

Comment:
I would separate the key and val sizes in the message... something like the following with a space between... gives a little more info.

```
throw new IllegalArgumentException("Key/value pair is too large (" + ((long) key.getSize() +" " + (long) value.getSize())
```